tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -211,6 +213,78 @@ class Restrictions {
     return [];
   }
 
+  List<SourceSpanSeverityException> validateExtendsClassKey(
+    String parentNodeName,
+    dynamic extendsClass,
+    SourceSpan? span,
+  ) {
+    if (extendsClass is! String) {
+      return [
+        SourceSpanSeverityException(
+          'The "${Keyword.extendsClass}" key must be a String.',
+          span,
+        )
+      ];
+    }
```

Comment:
Suggestion: With the previously mentioned change of removing the isExperimentalFeatureEnabled from this method, I think the entire method can be deleted as we don't need any key specific validation anymore.